src/os/bluestore/BlueStore.cc

```diff
@@ -1708,9 +1706,15 @@ bool BlueStore::ExtentMap::update(KeyValueDB::Transaction t,
     }
     //schedule DB update for dirty shards
     auto it = dirty_shards.begin();
+    bool first_key = true;
```

Comment:
You might want to replace this and other duplicate code that enumerates over dirty shards with a single function that additionally accepts a function pointer (lambda) to fulfill custom action per each key, e.g. call t->set or t->rmkey